### DISCUSSION TITLE
Assessment of the difficulty in porting CPU architecture for PEGTL

### DISCUSSION THREAD

**Discussion by wangyuliu** [taocpp/PEGTL#355] [General]
Hello everyone! I am working on implementing a tool to assess the complexity of CPU architecture porting. It primarily focuses on RISC-V architecture porting. In fact, the tool may have an average estimate of various architecture porting efforts.My focus is on the overall workload and difficulty of transplantation in the past and future,even if a project has already been ported.As part of my dataset, I have collected the **PEGTL** project. **I would like to gather community opinions to support my assessment. I appreciate your help and response!** Based on scanning tools, the porting complexity is determined to be moderate leaning towards simple, with a small amount of code related to the CPU architecture in the project.  Is this assessment accurate? I look forward to your help and response.

**Comment by ColinH** [comment]
The PEGTL is essentially processor architecture independent and I would expect it to compile on/for RISC-V using GCC or Clang without modifications. The complexity of "porting" is therefore less than moderate or simple, it should be trivial, at least for RV64G.
